src/main/webapp/definitions/alfresco-core.yaml

```diff
@@ -10690,6 +10786,30 @@ definitions:
         $ref: '#/definitions/PermissionsInfo'
       definition:
         $ref: '#/definitions/Definition'
+  calculateSizeResponse:
+    type: object
+    properties:
+      id:
+        type: string
+        description: Provides the id for which POST/calculateSize has been initiated.
```

Comment:
This is a little bit weird. For consistency I think this should be the id of the background job, not the node. If the background job does not have an id then this is ok.

Reply:
We provide the **nodeId** in the POST Endpoint response instead **backgroundJobId** because it's removed once the action is completed. As a result, we cannot discover the size in the GET Endpoint response.

Reply:
The pattern in most places in the v1 API is:

* `POST /parents/{parent-id}/children` returns `{"id": "child-id", ...}`
* `GET /parents/{parent-id}/children` returns paged response including `{"id": "parent-id", ...}`
* `GET /parents/{parent-id}/children/{child-id}` returns `{"id": "parent-id", ...}`

Here instead we are doing:

* `POST /parents/{parent-id}/action` returns `{"id": "parent-id", ...}`
* `GET /parents/{parent-id}/action` returns `{"id": "parent-id", ...}` (not paged)

Maybe this is consistent with `/nodes/{node-id}/move` and `/nodes/{node-id}/copy` though, where we're really performing an action on the resource, rather than a creation.

Reply:
We are also referring to the following V1 format:
However, in the case of POST, we return the parent-id rather than the **execution job ID.**

- `POST /nodes/{node-id}/calculateSize` returns {"id": "child-id",...}.

For GET, we return a response with the parent-id (not paged).

- `GET /nodes/{node-id}/calculateSize` returns {"id": "parent-id",...}.

Note: In the preceding comment, **"action"** referred to the Action executor used to execute tasks asynchronously, not any endpoint. For better clarity and understanding, we may set up a call.

Reply:
Now, we are considering an execution job ID instead of a node ID and returning the execution job ID as a response in POST endpoint.